arroyo/processing/run_task.py

```diff
@@ -0,0 +1,77 @@
+import logging
+import time
+from collections import deque
+from concurrent.futures import Future, ThreadPoolExecutor
+from typing import Callable, Deque, Optional, Tuple
+
+from arroyo.processing.strategies.abstract import MessageRejected, ProcessingStrategy
+from arroyo.types import Commit, Message, Position, TPayload
+
+logger = logging.getLogger(__name__)
+
+
+class RunTask(ProcessingStrategy[TPayload]):
+    """
+    Runs a task in a thread. Commits offsets once futures are done.
```

Comment:
Do we have a scenario for this kind of consumer?

I think our common scenarios are:
- transformer: take a message, transform it and produce it somewhere 
- map-reduce: take a message, transform it, accumulate the result in a batch, periodically write.
- batch-transform-unbatch: accumulate messages, run a processing on batches, unbatch the results and produce on a topic.

All of them do something with the results of the processing.